src/diffusers/models/attention.py

```diff
@@ -58,6 +58,7 @@ def __init__(
         only_cross_attention: bool = False,
         double_self_attention: bool = False,
         upcast_attention: bool = False,
+        upcast_softmax: bool = False,
```

Comment:
What is this needed for?

Reply:
the original repo did that - it upcast to float32 for the softmax step during attention calculation
I didn't notice any issue with fp16 otherwise